lib/migrator.js

```diff
@@ -6,53 +6,158 @@ const { moveFile, removeDirs } = require('./utils/file')
+      const filePathFromAppTemplatesComponents = filePathFromApp.slice('app/templates/components/'.length);
 
-    templatesWithLayoutName.sort().forEach(template => {
-      console.info(`❌ Did not move '${template}' due to usage as "layoutName" in a component`);
+      // '[APP_PATH]/app/components/nested1/nested-component.hbs'
+      const newTemplateFilePath = path.join(this.projectRoot, 'app/components', filePathFromAppTemplatesComponents);
+      moveFile(templateFilePath, newTemplateFilePath);
     });
-    templatesInPartials.sort().forEach(template => {
-      console.info(`❌ Did not move '${template}' due to usage as a "partial"`);
+  }
+
+  changeComponentStructureToNested(templateFilePaths) {
+    const classFilePaths = this.findClassicComponentClasses();
+
+    templateFilePaths.forEach(templateFilePath => {
+      // Extract '/app/templates/components/nested1/nested-component.hbs'
+      const filePathFromApp = templateFilePath.slice(this.projectRoot.length);
+
+      // Extract '/nested1/nested-component.hbs'
+      const filePathFromAppTemplatesComponents = filePathFromApp.slice('app/templates/components/'.length);
+      const fileExtension = path.extname(filePathFromAppTemplatesComponents);
+
+      // Extract '/nested1/nested-component'
+      const targetPath = filePathFromAppTemplatesComponents.slice(0, -fileExtension.length);
+
+      // Build '[APP_PATH]/app/components/nested1/nested-component/index.hbs'
+      const newTemplateFilePath = path.join(this.projectRoot, 'app/components', targetPath, 'index.hbs');
+      moveFile(templateFilePath, newTemplateFilePath);
+
+      // Build '[APP_PATH]/app/components/nested1/nested-component/index.js'
+      const classFilePath = path.join(this.projectRoot, 'app/components', `${targetPath}.js`);
+
+      if (classFilePaths.includes(classFilePath)) {
+        const newClassFilePath = path.join(this.projectRoot, 'app/components', targetPath, 'index.js');
+        moveFile(classFilePath, newClassFilePath);
+      }
     });
+  }
+
+  async removeEmptyClassicComponentDirectories() {
+    const templateFolderPath = path.join(this.projectRoot, 'app/templates/components');
+
+    const classFilePaths = this.findClassicComponentClasses();
+    const templatesWithLayoutName = getLayoutNameTemplates(classFilePaths);
+    const removeOnlyEmptyDirectories = Boolean(templatesWithLayoutName.length);
+ 
+    await removeDirs(templateFolderPath, removeOnlyEmptyDirectories);
+  }
+
+  async execute() {
+    let templateFilePaths = this.findClassicComponentTemplates();
+    templateFilePaths = this.skipTemplatesUsedAsLayoutName(templateFilePaths);
+    templateFilePaths = this.skipTemplatesUsedAsPartial(templateFilePaths);
+
+    if (this.newComponentStructure === 'flat') {
+      this.changeComponentStructureToFlat(templateFilePaths);
+
+    } else if (this.newComponentStructure === 'nested') {
+      this.changeComponentStructureToNested(templateFilePaths);
+
+    }
```

Comment:
FWIW, I'd like to make another structure (not sure what a good name for it would be) to use flat when there is only a template or class file and nested when there are both (e.g. to avoid `app/components/foo.js` and `app/components/foo.hbs` living side by side). 

This is definitely out of scope for your PR though, just thinking out loud... 🤔